### PR TITLE
[#48] Set up build variants

### PR DESCRIPTION
**Pull request by chornerman**
#48 

## What happened 👀

We need to set up application build variants for both staging and production

## Insight 📝

Set up application build variants

- [x] Set up **Staging** flavor
  - The app name should be "Survey Staging" 
  - The package name should be `co.nimblehq.survey.staging`
  - The rest API endpoint should be `https://nimble-survey-web-staging.herokuapp.com/`
- [x] Set up **Production** flavor
  - The app name should be "Survey" 
  - The package name should be `co.nimblehq.survey`
  - The rest API endpoint should be `https://survey-api.nimblehq.co/`

## Proof Of Work 📹

https://user-images.githubusercontent.com/13492460/170301495-877a5923-de7f-4686-ac8c-c8ab54a69eef.mp4

### CODE REVIEW

File: lib/main_production.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+
+import 'app_config.dart';
+import 'main.dart';
+
+void main() {
+  var configuredApp = const AppConfig(
+    appName: 'Survey',
+    flavorName: 'production',
+    endpoint: 'https://survey-api.nimblehq.co/',
```
Comment (markgravity) [inline lib/main_production.dart:10]: I'm not sure how we access this value in the network service? 🤔

Comment (chornerman) [reply lib/main_production.dart]: @markgravity Updated. Please help take a look again 🙏🏼

File: lib/flavors.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        return 'Survey Staging';
+    }
+  }
+
+  static String get apiEndpoint {
+    switch (appFlavor) {
+      case Flavor.production:
+        return 'https://survey-api.nimblehq.co/';
+      default:
+        return 'https://nimble-survey-web-staging.herokuapp.com/';
```
Comment (markgravity) [inline lib/flavors.dart:23]: What do you think about removing the `/` at the end of the URL?
We will add it when we use it 🤔 
```dart
"${F.apiEndPoint}/login"

**Review by markgravity — COMMENTED** [review]
The rest LGTM

File: lib/flavors.dart (hunk 2 above)
Comment (chornerman) [reply lib/flavors.dart]: Resolved 😄

**Review by markgravity — APPROVED** [review]
LGTM 👍

File: android/app/build.gradle [hunk 3]
```diff
@@ -50,6 +50,21 @@ android {
...
+    productFlavors {
+        production {
+            dimension "stage"
+            applicationId "co.nimblehq.survey"
+            resValue "string", "app_name", "Survey"
+        }
+        staging {
+            dimension "stage"
+            applicationId "co.nimblehq.survey.staging"
+            resValue "string", "app_name", "Survey Staging"
```
Comment (luongvo) [inline android/app/build.gradle:64]: @chornerman we should use strings.xml in each flavor to define app names for each env instead

**Review by luongvo — COMMENTED** [review]
rest lgtm @chornerman

File: android/app/build.gradle (hunk 3 above)
Comment (chornerman) [reply android/app/build.gradle]: @luongvo Updated 👍🏼

**Review by luongvo — APPROVED** [review]
lgtm

**Pull request merged**